pkg/apis/admissionregistration/types.go

```diff
@@ -1163,3 +1164,317 @@ type MatchCondition struct {
 	// Required.
 	Expression string
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:prerelease-lifecycle-gen:introduced=1.32
+
+// MutatingAdmissionPolicy describes an admission policy that may mutate an object.
+type MutatingAdmissionPolicy struct {
```

Comment:
Since we have parameters, don't we need the status to represent errors that are not resolvable until the parameter types are known?

Reply:
OK if we add this before beta but not in 1.32? I have a branch that can do the compilation correctly but it's more complicated than VAP due to the new types introduced.